environment/environment.php

```diff
@@ -115,21 +115,18 @@ public function __construct(
 	/**
 	 * Creates the environment based on the linkItem the token links to
 	 *
-	 * @param array $linkItem
+	 * @param object $linkItem
 	 */
 	public function setTokenBasedEnv($linkItem) {
-		// Resolves reshares down to the last real share
-		$rootLinkItem = Share::resolveReShare($linkItem);
-		$origShareOwner = $rootLinkItem['uid_owner'];
-		$this->userFolder = $this->rootFolder->getUserFolder($origShareOwner);
-
-		// This is actually the node ID
-		$this->sharedNodeId = $linkItem['file_source'];
-		$this->fromRootToFolder = $this->buildFromRootToFolder($this->sharedNodeId);
-
-		$this->folderName = $linkItem['file_target'];
-		$this->userId = $rootLinkItem['uid_owner'];
-		$this->sharePassword = $linkItem['share_with'];
+        $origShareOwnerId = $linkItem->getShareOwner();
+        $this->userFolder = $this->rootFolder->getUserFolder($origShareOwnerId);
+
+        $this->sharedNodeId = $linkItem->getNodeId();
+        $this->fromRootToFolder = $this->buildFromRootToFolder($this->sharedNodeId);
+
+        $this->folderName = $linkItem->getTarget();
+        $this->userId = $origShareOwnerId;
```

Comment:
I think this may be wrong. There is a difference between the share owner and the file owner